### PR TITLE
show prompt before file download, mic and camera access

### PR DESCRIPTION
**Pull request by jpcloureiro**

### CODE REVIEW

**Comment by jpcloureiro** [comment]
free to merge once approved

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebChromeClient.java [hunk 1]
```diff
@@ -169,9 +174,23 @@ public void onPermissionRequest(final PermissionRequest request) {
...
+                    builder.setPositiveButton("Allow", (dialog, which) -> {
+                        permissionRequest = request;
+                        grantedPermissions.add(finalAndroidPermission);
+                        requestPermissions(grantedPermissions);
+                    });
+                    builder.setNegativeButton("Don't allow", (dialog, which) -> {
+                        request.deny();
+                    });
+                    AlertDialog alertDialog = builder.create();
+                    alertDialog.show();
```
Comment (tommasini) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebChromeClient.java:193]: The delayed function to accept this permissions was implemented here as well?
Would something like this worth the try?
```
this.runDelayed(() -> builder.setPositiveButton("Allow", (dialog, which) -> {
                        permissionRequest = request;
                        grantedPermissions.add(finalAndroidPermission);
                        requestPermissions(grantedPermissions);
                    }); , 500);

```

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManagerImpl.kt [hunk 2]
```diff
@@ -106,34 +108,42 @@ class RNCWebViewManagerImpl {
...
+                        )
+                ) {
+                    module.downloadFile(
+                            getDownloadingMessageOrDefault()
+                    )
+                }
             }
+            builder.setNegativeButton("Cancel") { _: DialogInterface?, _: Int -> }
+            val alertDialog = builder.create()
+            alertDialog.show()
```
Comment (tommasini) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManagerImpl.kt:146]: For consistency should we have the delay function here as well? I think here it wasn't at the original patch, just wondering

**Comment by jpcloureiro** [comment]
Thanks for pointing that @tommasini 
The delay implementation is taken care on another PR to keep changes history consistent with what we have on `react-native-webview+11.13.0.patch`

**Pull request merged**